lib/network/kmip/ClusterClient.ts

```diff
@@ -56,25 +90,168 @@ export default class ClusterClient implements KMSInterface {
         return this.clients[this.roundRobinIndex++];
     }
 
+    checkUnhealthyClient(unhealthyClient: UnhealthyClient) {
+        const client = unhealthyClient.client;
+        const healthyIndex = this.clients.indexOf(client);
+        if (healthyIndex === -1) {
+            this.clients.push(client);
+        } else {
+            // should not happen
```

Comment:
If this should not happen, why do we have a condition for this?

Reply:
Okay, wondering why we need defensive programming here, if it will never happen.
Do you see cases where it might happen(for my understanding). I am okay with this generally.

Reply:
Let's add the ticket number for the"later" part so people have context from comments

Reply:
Added a ticket number [a59c1cd](https://github.com/scality/Arsenal/pull/2397/commits/a59c1cdd234430eaa685563c94c11127fd109dd3)